src/edits.ts

```diff
@@ -28,89 +28,113 @@ export interface EditOptions {
 }
 
 export async function uploadToPlayStore(options: EditOptions, releaseFiles: string[]): Promise<string | undefined> {
-    const appEdit = await androidPublisher.edits.insert({
-        auth: options.auth,
-        packageName: options.applicationId
-    });
-    
-    releaseFiles.forEach(async releaseFile => {
-        core.debug(`Uploading ${releaseFile}`)
-        await uploadRelease(appEdit.data, options, releaseFile).catch(reason => {
-            return Promise.reject(reason)
-        })
-    })
-
-    const res = await androidPublisher.edits.commit({
-        auth: options.auth,
-        editId: appEdit.data.id!,
-        packageName: options.applicationId
-    });
-
-    if (res.data.id != null) {
-        core.debug(`Successfully committed ${res.data.id}`)
-        return Promise.resolve(res.data.id!)
-    } else {
-        core.setFailed(`Error ${res.status}: ${res.statusText}`)
-        return Promise.reject(res.status)
-    }
-}
-
-async function uploadRelease(appEdit: AppEdit, options: EditOptions, releaseFile: string): Promise<string | undefined | null> {
     // Check the 'track' for 'internalsharing', if so switch to a non-track api
-    if (options.track === 'internalsharing'){
+    if (options.track === 'internalappsharing') {
         core.debug("Track is Internal app sharing, switch to special upload api")
-        if (releaseFile.endsWith('.apk')) {
-            const res = await internalSharingUploadApk(options, releaseFile)
-            return Promise.resolve(res.downloadUrl)
-        } else if (releaseFile.endsWith('.aab')) {
-            const res = await internalSharingUploadBundle(options, releaseFile)
-            return Promise.resolve(res.downloadUrl)
+        releaseFiles.forEach(async releaseFile => {
+            core.debug(`Uploading ${releaseFile}`);
+            await uploadInternalSharingRelease(options, releaseFile).catch(reason => {
+                core.setFailed(reason);
+                return Promise.reject(reason);
+            });
+        });
+    } else {
+        // Create a new Edit
+        const appEdit = await androidPublisher.edits.insert({
+            auth: options.auth,
+            packageName: options.applicationId
+        });
+        
+        // Validate the given track
+        await validateSelectedTrack(appEdit.data, options).catch(reason => {
+            core.setFailed(reason);
+            return Promise.reject(reason);
+        });
+    
+        // Upload artifacts to Google Play, and store their version codes
+        var versionCodes = new Array<number>();
+        releaseFiles.forEach(async releaseFile => {
```

Comment:
You are going to run into issues with this block (and similar others) where you are looping and spinning off async tasks without blocking for them so the next blocks of code (add releases, committing edit) are going to potentially execute without this finishing. You should refer to this post (https://stackoverflow.com/a/37576787) for writing these loops in a blocking fashion.

Reply:
This sounds like what's causing the issue I mentioned earlier, and would explain why it sometimes works. Thanks for catching that!